src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown

```diff
@@ -0,0 +1,319 @@
+require HTTPS. The WebUSB API is no different, and is yet another good reason
+to get HTTPS up and running on your site.
+
+During development you'll be able to interact with WebUSB through
+http://localhost by using tools like the [Chrome Dev
+Editor](https://chrome.google.com/webstore/detail/pnoffddplpippgcfjdhbmhkofpnaalpg)
+or the handy `python -m SimpleHTTPServer`, but to deploy it on a site you'll
+need to have HTTPS set up on your server. I personally enjoy [GitHub
+Pages](https://pages.github.com) for demo purposes.
+
+To add HTTPS to your server you'll need to get a TLS certificate and set
+it up. Be sure to check out the [Security with HTTPS
+article](/web/fundamentals/security/)
+for best practices there. For info, you can now get free TLS certificates with
+the new Certificate Authority [Let's Encrypt](https://letsencrypt.org/).
+
+### User Gesture Required
+
+As a security feature, getting access to connected USB devices with
+`navigator.usb.requestDevice` must be called via a user gesture
+like a touch or mouse click.
+
+## Let's start coding
+
+The WebUSB API relies heavily on JavaScript
+[Promises](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+If you're not familiar with them, check out this great
+[Promises tutorial](http://www.html5rocks.com/en/tutorials/es6/promises/). One
+more thing, `() => {}` are simply ECMAScript 2015 [Arrow
+functions](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+-- they have a shorter syntax compared to function expressions and lexically
+bind the value of `this`.
+
+### Get Access to USB devices
+
+You can either prompt the user to select a single connected USB device using
+`navigator.usb.requestDevice` or call `navigator.usb.getDevices` to get a list
+of all connected USB devices the origin has access to.
+
+The `navigator.usb.requestDevice` function takes a mandatory JavaScript object
+that defines `filters`. These filters are used to match any USB device with the
+given vendor (`vendorId`) and optionally product (`productId`) identifiers. The
+`classCode`, `subclassCode` and `protocolCode` keys can also be defined there
+as well.
+
+<img style="width:723px; max-height:414px" src="/web/updates/images/2016-03-02-access-usb-devices-on-the-web/usb-device-chooser.png" alt="USB Device Chooser screenshot"/>
+
+For instance, here's how to get access to a connected Arduino device configured
+to allow the origin.
+
+{% highlight javascript %}
+navigator.usb.requestDevice({ filters: [{ vendorId: 0x2341 }] })
+.then(device => {
+  console.log(device.productName);      // "Arduino Micro"
+  console.log(device.manufacturerName); // "Arduino LLC"
+})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+Before you ask, I didn't magically come up with this `0x2341` hexadecimal
+number. I simply searched for the word "Arduino" in the official [List of USB
```

Comment:
This list actually isn't official. It's just something the Linux USB subsystem maintainers put together.